docs/resources/task.md

```diff
@@ -0,0 +1,224 @@
+- `timeout` - (Optional) Maximum number of seconds to run before termination.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+- `ssh_public_key` - Used to access the created machines.
+- `ssh_private_key` - Used to access the created machines.
+- `addresses` - IP addresses of the currently active machines.
+- `status` - Status of the machine orchestrator.
+- `events` - List of events for the machine orchestrator.
+- `logs` - List with task logs; one for each machine.
+
+~> **Note** Status and events don't produce a stable output between cloud providers and are intended for human consumption only.
+
+## Machine Type
+
+### Generic
+
+The Iterative Provider offers some common machine sizes which are roughly the same
+for all supported clouds.
+
+#### No GPU
+
+- `m` - Medium, with (at least), 8 CPU cores and 16 GB of RAM.
+- `l` - Large, with (at least) 32 CPU cores and 64 GB of RAM.
+- `xl` - Extra large, with (at least) 64 CPU cores and 128 GB of RAM.
+
+#### NVIDIA Tesla K80 GPU
+
+- `m+k80` - Medium, with (at least) 4 CPU cores, 53 GB of RAM and 1 GPU device.
+- `l+k80` - Large, with (at least) 12 CPU cores, 112 GB of RAM and 2 GPU devices.
+- `xl+k80` - Extra large, with (at least) 24 CPU cores, 212 GB of RAM and 4 GPU devices.
+
+#### NVIDIA Tesla V100 GPU
+
+- `m+v100` - Medium, with (at least) 4 CPU cores, 61 GB of RAM and 1 GPU device.
+- `l+v100` - Large, with (at least) 12 CPU cores, 224 GB of RAM and 2 GPU devices.
+- `xl+v100` - Extra large, with (at least) 24 CPU cores, 448 GB of RAM and 4 GPU devices.
+
+### Cloud-specific
+
+In addition to generic types, it's possible to specify any machine type
+supported by the underlying cloud provider.
+
+#### Amazon Web Services
+
+- `{machine}` - Any [EC2 instance type](https://aws.amazon.com/ec2/instance-explorer) (e.g. `g4dn.xlarge`).
+
+#### Google Cloud Platform
+
+- `{machine}` - Any [GCP machine type](https://cloud.google.com/compute/docs/machine-types) (e.g. `n2-custom-64-262144`).
+- `{machine}+{accelerator}*{count}` - Any machine and accelerator combination (e.g. `custom-8-53248+nvidia-tesla-k80*1`).
+
+#### Microsoft Azure
+
+- `{machine}` - Any [Azure VM](https://azure.microsoft.com/en-us/pricing/vm-selector) (e.g. `Standard_F8s_v2`).
+
+#### Kubernetes
+
+- `{cpu}-{memory}` - Any [CPU & memory combination](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers) (e.g. `64-256000`).
+- `{cpu}-{memory}+{accelerator}*{count}` - Any CPU, memory, & [accelerator](https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus) combination (e.g. `64-256000+nvidia-tesla-k80*1`).
+
+-> **Note** Specified resource amounts are considered **limits** (rather than **requests**).
+
+-> **Note** `{accelerator}` will be transformed into a node selector requesting `accelerator={accelerator}` and `{count}` will be configured as the **limits** count for `kubernetes.io/gpu`.
```

Comment:
```suggestion
> **Note**: Specified resource amounts are considered **limits** (rather than
> **requests**).

> **Note**: `{accelerator}` will be transformed into a node selector requesting
> `accelerator={accelerator}` and `{count}` will be configured as the **limits**
> count for `kubernetes.io/gpu`.
```